### PR TITLE
#2545: fix hang in "frequent" suite with slow dispatch

### PR DESCRIPTION
**Pull request by TT-billteng**
This required migrating pytest in test/models folder to use `device` pytest fixture. Also, a clean up step to delete allocated buffers after each test is done has been added. Using the `device` pytest fixture exposed these memory leaks.

### CODE REVIEW

**Review by tt-rkim — APPROVED** [review]
lgtm, would like @tt-aho 's feedback

**Review by tt-aho — COMMENTED** [review]
LGTM. My code comments are just related to formatting and removing the missed `if __name__ == "__main__":` blocks.
Please also fix your first commit message which is missing #.

On another note, do we know what these memory leaks are? ie. are they Tensors? Adding this deallocate at the end of each test would essentially hide any new or existing memory leaks right?

**Review by ashayestehmanesh — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]
I agree my changes are low priority and can be done in separate pr if we want. Please fix commit msg before merge though. Meant to approve before to unblock but realized I only commented. Approving now

**Pull request merged**
